homeassistant/components/xbox/__init__.py

```diff
@@ -115,19 +120,46 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
         )
     )
     if unload_ok:
+        # Unsub from coordinator updates
+        hass.data[DOMAIN][entry.entry_id]["binary_sensor_unsub"]()
         hass.data[DOMAIN].pop(entry.entry_id)
 
     return unload_ok
 
 
 @dataclass
-class XboxData:
-    """Xbox dataclass for update coordinator."""
+class ConsoleData:
```

Comment:
Any reason to rename this dataclass when adding a new platform?

Reply:
I needed access to the previous state which is not available inside the callback. I’m not renaming it, I’m extending it and overloading the update method.

Reply:
Whoops, scratch that. Didn’t have my caffeine yet.

That was renamed because I now store both Xbox Console and User Presence data as a part of the update. The base class that stores both is now XboxData, and I renamed what was XboxData (which only stores console info) to ConsoleData.